src/Perl6/Optimizer.nqp

```diff
@@ -1916,6 +1916,12 @@ class Perl6::Optimizer {
               elsif $sigil eq '$' {
                 $assignop := 'assign';
               }
+              elsif $sigil eq '@' {
+                $assignop := 'p6store';
+              }
+              elsif $sigil eq '%' {
+                $assignop := 'p6store';
+              }
```

Comment:
You can just write a single conditional as `elsif $sigil eq '@' || $sigil eq '%' {...}` and avoid duplication of the body.